repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/spring/CoreSpringProvider.java

```diff
@@ -163,31 +164,31 @@ public void initializeCoreContext(String configRoot,
             // I'm not sure how well this will work, maybe it will fix the problem.
             GenericBeanDefinition mBeanServer = new GenericBeanDefinition();
             mBeanServer.setBeanClass(MBeanServerFactoryBean.class);
-            mBeanServer.setAutowireCandidate(true);
-            mBeanServer.setLazyInit(false); //Make sure that it's not lazy!
             MutablePropertyValues mBeanServerProps = new MutablePropertyValues();
             //This should tell spring to ENSURE that it'll use an existing server
             mBeanServerProps.add("locateExistingServerIfPossible", true);
             mBeanServer.setPropertyValues(mBeanServerProps);
             //Would have to give this an agent id?
             coreContext.registerBeanDefinition("reposeMBeanServer", mBeanServer);
 
+            GenericBeanDefinition jmxAttributeSource = new GenericBeanDefinition();
+            jmxAttributeSource.setBeanClass(AnnotationJmxAttributeSource.class);
+            coreContext.registerBeanDefinition("jmxAttributeSource", jmxAttributeSource);
+
+            GenericBeanDefinition jmxNamingStrategy = new GenericBeanDefinition();
+            jmxNamingStrategy.setBeanClass(ReposeJmxNamingStrategy.class);
+            jmxNamingStrategy.setAutowireMode(AbstractBeanDefinition.AUTOWIRE_CONSTRUCTOR);
```

Comment:
This is the primary component that we were missing before. We defined the naming strategy as a bean, but never tied it into the `MBeanExporter`. As a result, we were always using a different naming strategy that the `MBeanExporter` defaulted to. To tie it in, I programmatically define the bean (for the sake of consistency and to prevent the previously mentioned tests from breaking) and set it on the `MBeanExporter`.